neurodamus/core/coreneuron_configuration.py

```diff
@@ -80,63 +80,136 @@ class _CoreNEURONConfig:
     Note: this creates the `CoreConfig` singleton
     """
 
-    sim_config_file = "sim.conf"
-    report_config_file = "report.conf"
-    restore_path = None
-    output_root = "output"
-    datadir = f"{output_root}/coreneuron_input"
     default_cell_permute = 0
     artificial_cell_object = None
 
+    @property
+    def sim_config_file(self):
+        """Get sim config file path to be saved"""
+        return str(Path(self.save_path) / "sim.conf")
```

Comment:
I prefer that`self.save_path` is not used for the normal workflow which is without `--save` even though it has the default value of output_root. Or you can rename it. The same thing for `SimConfig.coreneuron_datadir_save`.
IMO, attributes named as "_save" or "_restore" are better to be used only for --save or --restore. Otherwise, it is a bit confusing to read the code.

Reply:
the restore is used only for restore, no problem there already.

the save part is more tricky because I cannot implement over and over the logic of "use save if available otherwise use output" everywhere and save has priority... What do you think about `save_or_output_path`? It is longer tho

Reply:
```
"use save if available otherwise use output"
```
Why not?  It would appear only in the getter functions and it is clearer in logic. Better than `save_or_output_path`.

Reply:
because I need to implement the same logic many times and I prefer to have a single point of failure. If you prefer another name for the function I am open to suggestions but I do not think writing the same logic many times good practice when it can be stuffed in a function/property

Reply:
I need to puth this logic in a function. I do not know how to name this better than: `save_path` or `save_or_output_path`. If you have better ideas I am happy to hear them but I cannot reimplement the same logic over and over in all the writers. Advantages:

- Consistency – avoids duplicated logic across the codebase
- Maintainability – easy to update logic in one place
- Readability – method name shows intent clearly
- Testability – logic can be unit tested in isolation
- Error reduction – minimizes risk of inconsistent or incorrect usage

Reply:
or eval_tstop()

Reply:
done, `generated_ouput_path`. I do not like it very much but my main concern is not about the name. I am happy to have the function